lib/Parser/Parse.cpp

```diff
@@ -7292,6 +7275,11 @@ LPCOLESTR Parser::GetFunctionName(ParseNodeFnc * pnodeFnc, LPCOLESTR pNameHint)
     {
         name = pNameHint;
     }
+    if ((name == nullptr && pnodeFnc->IsLambda()) ||
```

Comment:
What if the name is non-null and the function is not a declaration?

Reply:
Good point, I added these parens to fix xplat build warnings but looks like I really wanted this:
```
    if (name == nullptr && (pnodeFnc->IsLambda() ||
        (!pnodeFnc->IsDeclaration() && !pnodeFnc->IsMethod())))
```

Reply:
What I'm asking is, why do we want to treat a named function expression as not having a name?

Reply:
Won't we have `pnodeFnc->pnodeName` if the function expression is named? Failing that, we would have `pNameHint` in case of expression assignment?

Reply:
I see. Yes, I didn't realize the parens expressed something other than what you intended. If name is null, is it necessary to check for these other cases? Under what conditions do we want to call the GetExternalDisplayName?